### PR TITLE
Allow site to specify using HUD chronic instead of potentially chronic

### PR DESCRIPTION
**Pull request by awisema**

### CODE REVIEW

File: app/controllers/concerns/site_chronic.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        when :chronic
+          :chronics
+        when :hud_chronic
+          :hud_chronics
+        else
+          raise NotImplementedError
+      end
+    end
+
+    def site_chronics_where(arg)
```
[the remark this answers is not in the document]
Comment (eanders) [reply app/controllers/concerns/site_chronic.rb]: After adding `site_chronics_table` I think we can remove the `site_chronics_where` method and just replace calls to it with `site_chronics_table => args` where args is whatever was being passed in.

File: app/controllers/warehouse_reports/cas/chronic_reconciliation_controller.rb [hunk 2]
```diff
@@ -2,19 +2,20 @@ module WarehouseReports::Cas
...
         where(ch_t[:date].eq(@filter.date)).
         where(ch_t[:days_in_last_three_years].gteq(365)).
         has_homeless_service_between_dates(start_date: @filter.homeless_service_after, end_date: @filter.date).
         pluck(:id)
 
       cas_ids = client_source.cas_active.pluck(:id)
-      @missing_in_cas = client_source.joins(:chronics).
-        where(chronics: {date: @filter.date}).
+      @missing_in_cas = client_source.joins(site_chronics_sym).
+        where(site_chronics_where({date: @filter.date})).
```
Comment (eanders) [inline app/controllers/warehouse_reports/cas/chronic_reconciliation_controller.rb]: See the note about `site_chronics_table` this becomes `where(site_chronics_table => {date: @filter_date})`

File: app/controllers/warehouse_reports/cas/chronic_reconciliation_controller.rb [hunk 3]
```diff
@@ -25,7 +26,7 @@ def index
       @not_on_list = client_source.
         where(id: (cas_ids - chronic_ids)).
         order(last_name: :asc, first_name: :asc).
-        includes(:chronics)
+        includes(site_chronics_sym)
```
Comment (eanders) [inline app/controllers/warehouse_reports/cas/chronic_reconciliation_controller.rb]: Any of these calls to `site_chronics_sym` that are in active record queries will probably be more readable with `site_chronics_table`

File: app/controllers/warehouse_reports/cas/chronic_reconciliation_controller.rb (hunk 2 above)
Comment (eanders) [reply app/controllers/warehouse_reports/cas/chronic_reconciliation_controller.rb]: Just to confuse this even more, this is even cleaner with `merge(site_chronic_source.on_date(date: @date)`

File: app/controllers/concerns/site_chronic.rb [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  include Chronic
+  include HudChronic
+
+  included do
+    def site_chronic_definition
+      GrdaWarehouse::Config.get(:chronic_definition).to_sym
+    end
+
+    def site_chronics
+      case site_chronic_definition
```
[the remark this answers is not in the document]
Comment (awisema) [reply app/controllers/concerns/site_chronic.rb]: Pesky little plurals... We could alias chronics/hud_chronics in their models, but I think I would rather leave it the way it is.

**Pull request merged**